.github/workflows/accept.yml

```diff
@@ -209,3 +225,12 @@ jobs:
       - name: Run tests
         run: npm test
         if: ${{ github.event.inputs.target-test }}
+
+      # Why Change Owner of Container Working Directory?
+
+      # the working directory is created by the runner and mounted to the container.
+      # container user is root and the runner is not a user in the container.
```

Comment:
Minor concern is we will require all containers ensure the default user is `root`, which probably is the case with existing containers but may want to highlight this assumption, for example if in the future we want to follow Dockerfile security best practices of setting a non-root default user.

Reply:
I just tried to explicitly set the working directory, by adding
```
        working-directory: /
``` 
which results in the following ownerships:
```
  whoami
  id
  ls -al
  shell: sh -e {0}
root
uid=0(root) gid=0(root) groups=0(root)
total 104
drwxr-xr-x    1 root root 4096 Jul 19 20:20 .
drwxr-xr-x    1 root root 4096 Jul 19 20:20 ..
```

Could you try if that does the trick in the case of your workflows to have npm run properly?

Reply:
@kuerbis-martin Did another run around the block trying combinations of `working-directory` (including as default for `run`). Nope. `working-directory: /` as above is relative to the runner. The checked-out code on the container is at `/__w/appoptics-bindings-node-dev/appoptics-bindings-node-dev` and this is where the ownership problem is. There is an alternative workaround to `chown`. Checkout the code to `path: somewhere` and then set `working-directory` to the appropriate path. It was deemed "more unfriendly".

@cheempz The default user as created by @actions is `root`. It's a "disposable" container used for several minutes on their platform, and we are only using this workaround until https://github.com/actions/runner/issues/1203 is fixed. I think we good.

Reply:
Agree that we're using these containers in a very disposable way so I'm not that concerned with them running as `root` right now.  I do want to clarify that it's not really GH actions setting what the default container user is, it is something defined by the image's Dockerfile and/or during docker container start.  For example, the https://hub.docker.com/r/bitnami/laravel/dockerfile specifically sets `USER bitnami` so if I ran this container in Actions and execute the `id` command, it shows:
> Run id
uid=1000(bitnami) gid=1000(bitnami) groups=1000(bitnami)

So my concern is if we somehow update our Dockerfiles (or happen to use a container) where the USER is set to non-root.

Reply:
> So my concern is if we somehow update our Dockerfiles (or happen to use a container) where the USER is set to non-root.

I think the approach with setting `path` and `working-directory` together would avoid this problem, wouldn't it?

Reply:
> > So my concern is if we somehow update our Dockerfiles (or happen to use a container) where the USER is set to non-root.
> 
> I think the approach with setting `path` and `working-directory` together would avoid this problem, wouldn't it?

There is currently a bug (feature) of `container` `options` that does not allow to override presets including `--workdir` (or cheeky `-w`). Hence `working-directory` will have to be specified explicitly for each and every run command.

Reply:
> For example, the https://hub.docker.com/r/bitnami/laravel/dockerfile specifically sets `USER bitnami` so if I ran this container in Actions and execute the `id` command, it shows:
> 
> > Run id
> > uid=1000(bitnami) gid=1000(bitnami) groups=1000(bitnami)
> 
> So my concern is if we somehow update our Dockerfiles (or happen to use a container) where the USER is set to non-root.

Run test case in a container:
```
on: 
  workflow_dispatch:

jobs:
  job1:
    runs-on: ubuntu-latest 
    container:
      image: bitnami/laravel

    steps:
      - run: id

      - name: Checkout ${{ github.ref }}
        uses: actions/checkout@v2
```
>Run id
uid=1000(bitnami) gid=1000(bitnami) groups=1000(bitnami)

and...
> Run actions/checkout@v2
/usr/bin/docker exec  bce3d81fe68323cf11d01394f89816d2156bf87576fef30b1c581b6e4b5009a1 sh -c "cat /etc/*release | grep ^ID"
Syncing repository: ronilan/gh-actions-playground
Getting Git version info
Deleting the contents of '/__w/gh-actions-playground/gh-actions-playground'
The repository will be downloaded using the GitHub REST API
To create a local Git repository instead, add Git 2.18 or higher to the PATH
Downloading the archive
Writing archive to disk
Error: EACCES: permission denied, open '/__w/gh-actions-playground/gh-actions-playground/e8d6b584-2bcd-4a26-b0b5-ba362553b227.tar.gz'

I'll formalized this and open another @actions issue for that.
Something along the lines of: "checkout fails in a container where image has predefined a user"

Reply:
Issue is opened (https://github.com/actions/runner/issues/1214) and I think at this stage we can move on with *"root is moot"*.

Reply:
Hehe yes I see there've some other issues filed regarding this and [even this bit Docker actions needing to always run as root](https://docs.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions#user).  Let's just move on from my pointless point :)